spd/utils/wandb_utils.py

```diff
@@ -447,3 +449,21 @@ def wandb_setup(
             **({"Aggregated Report": report_url} if report_url else {}),
         },
     )
+
+
+_n_try_wandb_comm_errors = 0
+
+
+def try_wandb[**P, T](fn: Callable[P, T], *args: P.args, **kwargs: P.kwargs) -> T | None:
+    """Attempts to call `fn` and if it fails with a wandb CommError, logs a warning and returns
+    None. The choice of wandb CommError is to catch issues communicating with the wandb server but
+    not legitimate logging errors, for example not passing a dict to wandb.log, or the wrong
+    arguments to wandb.save."""
+    global _n_try_wandb_comm_errors
+    try:
+        return fn(*args, **kwargs)
+    except wandb.errors.CommError as e:
```

Comment:
I'm getting `wandb.errors.errors.AuthenticationError` sometimes, which I think we want to catch too. See [here](https://wandb.ai/goodfire/spd/runs/9hva17te/logs) (you have to download the file to look)